src/globus_sdk/experimental/auth_requirements_error/_validators.py

```diff
@@ -0,0 +1,65 @@
+from __future__ import annotations
+
+import typing as t
+
+from ._serializable import Serializable
+
+S = t.TypeVar("S", bound=Serializable)
+
+
+class ValidationError(ValueError):
+    pass
+
+
+def str_(name: str, value: t.Any) -> str:
+    if not isinstance(value, str):
+        raise ValidationError(f"'{name}' must be a string")
+    return value
+
+
+def opt_str(name: str, value: t.Any) -> str | None:
+    if value is None:
+        return None
+    if not isinstance(value, str):
+        raise ValidationError(f"'{name}' must be a string")
+    return value
+
+
+def opt_bool(name: str, value: t.Any) -> bool | None:
+    if value is None:
+        return None
+    if not isinstance(value, bool):
+        raise ValidationError(f"'{name}' must be a bool")
+    return value
+
+
+def str_list(name: str, value: t.Any) -> list[str]:
+    if not (isinstance(value, list) and all(isinstance(s, str) for s in value)):
+        raise ValidationError(f"'{name}' must be a list of strings")
+    return value
+
+
+def opt_str_list(name: str, value: t.Any) -> list[str] | None:
+    if value is None:
+        return None
+    if not (isinstance(value, list) and all(isinstance(s, str) for s in value)):
+        raise ValidationError(f"'{name}' must be a list of strings")
+    return value
+
+
+def opt_str_list_or_commasep(name: str, value: t.Any) -> list[str] | None:
+    if value is None:
+        return None
+    if isinstance(value, str):
+        value = value.split(",")
+    if not (isinstance(value, list) and all(isinstance(s, str) for s in value)):
+        raise ValidationError(f"'{name}' must be a list of strings")
```

Comment:
This is inaccurate for LegacyAuthorizationParameters and could be misleading if looking at the debug logs to determine why the parse failed.

```suggestion
        raise ValidationError(
            f"'{name}' must be a list of strings "
            "or a string of comma-delimited values."
        )
```

Reply:
Yeah, I waffled on this but can easily change. I believe that a test case assertion will change as well though. Let me cruise through Kurt's feedback and then I'll tackle this one.